manifests/rule/remote.pp

```diff
@@ -309,6 +309,51 @@
       }
     }
 
+    # Basic validation for the action queue parameters
+    if $queue_size {
+      # First check the queue size, which should not be less than 100 per the docs
+      if $queue_size < 100 {
+        # Warn the user about an errant configuration, but don't fail as RSyslog will still run when setup this way
+        notify { "Invalid queue_size specified for ${name}":
+          message  => "Action queue size for ${name}: ${queue_size} is less than 100 and can have adverse effects on RSyslog",
+          loglevel => 'warning',
+        }
+      }
+
+      # Check to ensure the low watermark is not defined higher than the queue size
+      if $queue_low_watermark {
```

Comment:
Everything below here should be a failure.

Also, I think that if any of these are specified, then the user should either be forced to specify `queue_size` or it should automatically be set to some reasonable value higher than `queue_high_watermark`.

Reply:
Actually, we should limit queue size to >= 100 in the parameters as well.

Reply:
I'm hesitant to put a fail() for the queue size as the official RSyslog docs don't specify a proper lower bound (they say roughly 100) and also the minimum supported value changes per release. Putting a hard limit adds a lot of technical debt to our module and would mean having to constantly verify against unpublished values by the upstream RSyslog code.

Reply:
@kendall-moore That makes sense. Can we either move to a `warn()` function as @op-ct suggested or make the notification optional?

Reply:
We definitely can. It would introduce some inconsistency within that defined type. Reference here: https://github.com/simp/pupmod-simp-rsyslog/blob/master/manifests/rule/remote.pp#L304-L307

I like the idea in this instance of the message occurring agent side at runtime instead of server side. The puppetserver log is very cluttered in large Puppet infrastructures and often times I find warning/deprecations go unnoticed as a result. It's also possible to force this down to debug or disable it using a resource collector, which while not ideal does provide the ability for users to suppress it if they indeed have a good reason to go lower on the queue size than the recommended minimum. It's not my call but as long as you and @op-ct have considered that I'm fine switching it.

Reply:
I think that either making it a server-side `warn()` or using `simplib::dlookup()` to provide a hiera-level disable option would be fine.